scrapers_next/hi/people.py

```diff
@@ -1,89 +1,128 @@
-        inputs = form.xpath(".//input")
-        # build list of all of the inputs of the form, clicking the button we need to click
-        data = {}
-        for inp in inputs:
-            name = inp.get("name")
-            value = inp.get("value")
-            inptype = inp.get("type")
-            if inptype == "submit":
-                if value == self.button_label:
-                    data[name] = value
-            else:
-                data[name] = value
+    soc_domain_and_handle_regex = re.compile(r"\.*\/*(\w+)\.com/(.+)")
 
-        # do second request
-        resp = scraper.post(self.url, data)
-        return resp
+    social_removal = [
+        "user",
+        "channel",
+        "playlists",
+        "photos",
+        "/",
+    ]
+    social_remove_regexes = [re.compile(string) for string in social_removal]
 
-    def __str__(self):
-        return f"FormSource('{self.url}', '{self.form_xpath}', '{self.button_label}')"
+    def process_item(self, item):
+        a_tag = item.xpath("a")[0]
+        member_page = a_tag.get("href")
+        member_photo = a_tag.xpath("img")[0].get("src")
 
+        member_text = a_tag.text_content().strip()
 
-class Legislators(HtmlListPage):
-    source = FormSource(
-        "https://www.capitol.hawaii.gov/members/legislators.aspx", "//form", "Show All"
-    )
-    selector = CSS("#ctl00_ContentPlaceHolderCol1_GridView1 tr")
-
-    LABELS = {
-        "first_name": "LabelFirst",
-        "party": "LabelParty",
-        "room": "LabelRoom2",
-        "voice": "LabelPhone2",
-        "fax": "LabelFAX2",
-        "email": "HyperLinkEmail",
-        "chamber": "LabelDis",
-        "district": "LabelDistrict",
-    }
+        # Conditional re-formats member_text in cases with multiple commas
+        #   Ex: Richards\r\n, III, Herbert M. "Tim" (D)
+        multi_commas = self.multi_commas_regex.search(member_text)
+        if multi_commas:
+            single_line = member_text.replace("\r\n", "")
+            comma_split = [x.strip() for x in single_line.split(",")]
+            member_text = f"{' '.join(comma_split[0:2])}, {comma_split[-1]}"
 
-    def process_item(self, item):
-        try:
-            link = CSS("a").match(item)[1]
-        except SelectorError:
-            self.skip()
-        data = {
-            "last_name": link.text_content(),
-            "url": link.get("href"),
-        }
-        for key, label in self.LABELS.items():
-            data[key] = CSS(f"[id$={label}]").match_one(item).text_content().strip()
-
-        party = {"(D)": "Democratic", "(R)": "Republican"}[data["party"]]
-        address = "Hawaii State Capitol, Room " + data["room"]
-        chamber = "upper" if data["chamber"] == "S" else "lower"
+        leader = self.leader_regex.search(member_text)
+        if leader:
+            name_parts = [x.strip() for x in leader.groups()]
+            # Extracts title: only leaders have one listed
+            title = name_parts.pop()
```

Comment:
Since title is declared in an if statement, the scraper fails out at line 107 since it's referencing a variable that may not have been declared. Could move this line down to that chunk and do an `if leader` check instead of `if title`

Reply:
Resolved in [newest commit](https://github.com/openstates/openstates-scrapers/pull/4247/commits/399cb4b2a6dff5f1160c4b686f77ecebc49576e1)